src/Illuminate/Validation/Rules/Unique.php

```diff
@@ -35,8 +35,8 @@ public function ignore($id, $idColumn = null)
             return $this->ignoreModel($id, $idColumn);
         }
 
-        $this->ignore = str_replace(',', '', $id);
-        $this->idColumn = str_replace(',', '', $idColumn ?? 'id');
+        $this->ignore = addslashes($id);
```

Comment:
Stupid q: is this escaping so the value can be literally used in a DB query, where no further escaping happens?

Reply:
Using `addslashes()` to escape a value for a DB query is quite dangerous.

Reply:
My thoughts, thus I'm asking. No very familiar with the exact code.

But I know at least for Postgres it doesn't even work.

Reply:
We aren't adding slashes for the DB query. We are adding slashes for the str_getcsv call.

Reply:
👍 thanks for clarification.

I think however that the code could use some comments as to why all these special handling.

Reply:
Because a vulnerability was reported. It was possible to perform a XSS using the ID

Reply:
It's not XSS.

Reply:
I know this is old article but i guess it tells you why addslashes is not perfect solution here - http://shiflett.org/blog/2006/addslashes-versus-mysql-real-escape-string

Reply:
Keep in mind that Taylor already clarified, this is **not** for escaping for the database but for another case. 

Also, the code already changed in the meantime and the `addslashes` was moved to the `__toString` and _only_ for the `ignore` and not the column.